File: ejercicio4/pwCharactersFilter.rb
require_relative 'filter'

class PwCharactersFilter
  include Filter

  def ejecutar(credenciales)
    if credenciales.password.count("! # $ % & ' ( ) * + , - . / : ; < = > ? @ [ \ ] ^ _ { | } ~`") > 0
      puts "Comprobación de caracteres especiales cumplida"
      return true
    else
      puts "Comprobación de caracteres especiales incorrecta. Por favor, añada algunos a su
      contraseña"
      return false
    end
  end
end

File: P2/Ejercicio1/lib/pwCharactersFilter.dart
import 'filter.dart';
import 'struct_credenciales.dart';

class PwCharactersFilter extends Filter {
  static const String _specialCharacters = "!#\$%&'()*+,-./:;<=>?@[\\]^_{|}~`";

  @override
  bool ejecutar(Credenciales credenciales) {
    for (int i = 0; i < credenciales.password.length; i++) {
      if (_specialCharacters.contains(credenciales.password[i])) {
        return true;
      }
    }
    return false;
  }

  @override
  String msgerror(){
    return "Comprobación de caracteres especiales incorrecta. Por favor, añada algunos a su contraseña.";
  }
}
